tutorials/holoscan_debugging/README.md

```diff
@@ -0,0 +1,286 @@
+set breakpoints and inspect application state as usual.
+
+#### I see a `gdb` Python Exception in the `gdb` log.
+
+`gdb` relies on a Python module for operations such as unwinding. If the Python module is not properly referenced in the container, you may see `gdb` errors appear in the console log such as the following:
+```bash
+"Python Exception <class 'NameError'>: Installation error: gdb._execute_unwinders function is missing"
+```
+To resolve the error, update your `PYTHONPATH` variable to include your GDB Python directory:
+```bash
+PYTHONPATH=${PYTHONPATH}:/usr/share/gdb/python
+```
+
+## Debugging a Python application with `pdb`
+
+### Background
+
+The [Python Debugger](https://docs.python.org/3/library/pdb.html) module is a built-in interactive debugging tool for Python programs. Similar to `gdb`, it supports setting breakpoints for interactive, headless, just-in-time debugging. You can use
+`pdb` to debug Holoscan SDK Python programs on any platform supported by Holoscan SDK.
+
+Holoscan SDK Python libraries serve as wrappers around Holoscan SDK C++ libraries. When debugging symbols are provided for
+C++ libraries, `pdb` can step into both core C++ libraries and Python wrapper code.
+
+### Prerequisites
+
+The `pdb` module is built in to modern Python versions. No additional installation is required.
+
+We will continue to use the __Holoscan SDK v2.3 development container__ from NGC for this section, which includes
+pre-installed versions of Python and Holoscan SDK.
+
+### Getting Started
+
+We will continue to debug the Endoscopy Tool Tracking application. The tutorial [`debug_pdb.sh`](./debug_pdb.sh) script
+is a self-contained example that will build the application and launch into a `pdb` debugging session.
+
+Run the script to get started:
+```sh
+./tutorials/holoscan_debugging/debug_pdb.sh
+```
+
+The script runs through the following steps:
+1. Builds the HoloHub container environment based on the Holoscan SDK NGC container.
+2. Builds the Endoscopy Tool Tracking __Python__ application in the container environment. By default we use the
+  `debug` build mode to generate detailed C++ debugging symbols for the Endoscopy Tool Tracking application.
+3. Launches the Endoscopy Tool Tracking application with `pdb`, which is invoked via the Python interpreter:
+```bash
+python3 -m pdb <command --args ...>
+```
+
+This command launches the Python version of the Endoscopy Tool Tracking application with a breakpoint set on the
+very first line. From this point you can set additional breakpoints, inspect application state, and control program
+execution.
+
+Although launched from Python, calls to `import holoscan` will load Python-wrapped C++ Holoscan SDK libraries
+along with their debugging symbols. You can set breakpoints in the corresponding Holoscan SDK C++ source code after each
+breakpoint launches. For instance:
+```py
+# Load the Holoscan core module in Python
+-> from holoscan.core import Application
```

Comment:
I'm a bit confused on this line, should the user enter this line into the (Pdb) prompt? If so, should we change it to:

```suggestion
(Pdb) from holoscan.core import Application
```
I did notice that if I hit `n` twice, I will see the following, so I'm not sure what to do here.
```
> /workspace/holohub/applications/endoscopy_tool_tracking/python/endoscopy_tool_tracking.py(19)<module>()
-> from holoscan.core import Application
```

Reply:
The line in question was output from Pdb to the console, but I see how it was confusing. I've reworked the example for clarity (hopefully).